### ISSUE TITLE
Cannot declare class srcDevDebugProjectContainerUrlMatcher, because the name is already in use

### ISSUE THREAD

**Issue by pyatnitsev** [symfony/symfony#25886]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 4.0.3

I have a issue with executing build-in command in Controller. 
I have following code:

`

        $application = new Application($this->get('kernel'));
        $application->setAutoExit(false);

        $input = new ArrayInput([
            'command' => 'cache:clear',
        ]);

        $output = new BufferedOutput();
        $application->run($input, $output);

        $content = $output->fetch();
`

It's throws `Compile Error: Cannot declare class srcDevDebugProjectContainerUrlMatcher, because the name is already in use` but in 3.4.x it was work

**Comment by nicolas-grekas** [comment]
Can you please try again with the latest version? If you still have the error, can you provide a reproducer?

**Comment by skiner68** [comment]
Same problem with 4.0.4 for me - exception is throw after cache:warmup is called :-(
```
protected function execute(InputInterface $input, OutputInterface $output) {
        $this->io = new SymfonyStyle($input, $output);

        // reset symfony cache
        $app = $this->getApplication();

        $input = new ArrayInput(array());
        $app->find('cache:clear')->run($input, $output);

        //$input = new ArrayInput(array());
        $app->find('cache:warmup')->run($input, $output);


        // reset runtime cache
       // $this->resetRuntimeCache();
        return 1;
 }
```

**Comment by nicolas-grekas** [comment]
The solution is to dump an anonymous class, like done for the container now, instead of a harcoded srcDevDebugProjectContainerUrlMatcher. That's something I tried in #25909 (with other changes), if you want to give it a try.

**Comment by TsulunAleksey** [comment]
Hi!

Nicolas, can you provide more details on your solution? In #25909 PR you mentioned, there are 28 files changed so it's not clear where the fix is.

Thank you.

**Comment by nicolas-grekas** [comment]
See both `Static*Dumper`, and [Router::getMatcher()](https://github.com/symfony/symfony/pull/25909/files#diff-570118d57208d33f3b9131fadcc12e3dR274-R322)

**Comment by TsulunAleksey** [comment]
Thank you!

Nicolas, if I need to clear just Route cache would it work if I just remove everything from .cache folder?

**Comment by nicolas-grekas** [comment]
You'll tell me if it works :)

**Comment by nicolas-grekas** [comment]
Simple fix embedded in #26626
